# Awesome bioinformatics tools

A curated list of awesome bioinformatics tools. Most of these tools are available on GitHub. 

## Sequence alignment

### DNA/RNA sequence alignment

minimap2: https://github.com/lh3/minimap2

pbmm2: https://github.com/PacificBiosciences/pbmm2

Winnowmap: https://github.com/marbl/Winnowmap

NGMLR: https://github.com/philres/ngmlr

BWA: https://github.com/lh3/bwa

BWA-MEM2: https://github.com/bwa-mem2/bwa-mem2

STAR: https://github.com/alexdobin/STAR

bowtie: https://github.com/BenLangmead/bowtie

bowtie2: https://github.com/BenLangmead/bowtie2

blastn: http://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/

### Protein sequence alignment

miniprot: https://github.com/lh3/miniprot

blastp: http://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/

## Processing of FASTA/BAM/VCF/BED files

Seqtk: https://github.com/lh3/seqtk

fastp: https://github.com/OpenGene/fastp

FastQC: https://github.com/s-andrews/FastQC

Samtools: https://github.com/samtools/samtools

HTSlib: https://github.com/samtools/htslib

HTSlib for Rust: https://github.com/rust-bio/rust-htslib

HTSlib for Java: https://github.com/samtools/htsjdk

Bcftools: https://github.com/samtools/bcftools

Bedtools: https://github.com/arq5x/bedtools2

## Genetic variation detection 

### SNP/Indel detection from short-read sequencing

Deepvariant: https://github.com/google/deepvariant

freebayes: https://github.com/freebayes/freebayes

GATK: https://github.com/broadinstitute/gatk

16GT: https://github.com/aquaskyline/16GT

Strelka2: https://github.com/Illumina/strelka

### SNP/Indel detection from long-read sequencing

Longshot: https://github.com/pjedge/longshot

PEPPER-Margin-DeepVariant: https://github.com/kishwarshafin/pepper

Clair3: https://github.com/HKU-BAL/Clair3

Clairvoyante: https://github.com/aquaskyline/Clairvoyante

NanoCaller: https://github.com/WGLab/NanoCaller

### Structural variation detection from short-read sequencing

manta: https://github.com/Illumina/manta

delly: https://github.com/dellytools/delly

LUMPY: https://github.com/arq5x/lumpy-sv

Pindel: https://github.com/genome/pindel

smoove: https://github.com/brentp/smoove

### Structural variation detection from long-read sequencing

CuteSV: https://github.com/tjiangHIT/cuteSV

Sniffles: https://github.com/fritzsedlazeck/Sniffles

SVision: https://github.com/xjtu-omics/SVision

pbsv: https://github.com/PacificBiosciences/pbsv

svim: https://github.com/eldariont/svim

nanosv: https://github.com/mroosmalen/nanosv

Picky: https://github.com/TheJacksonLaboratory/Picky

### Short Tandem Repeat (STR) detection

TRF: https://github.com/Benson-Genomics-Lab/TRF

#### STR detection from long reads

NanoRepeat: https://github.com/WGLab/Nanorepeat

RepeatHMM: https://github.com/WGLab/RepeatHMM

DeepRepeat: https://github.com/WGLab/DeepRepeat

Straglr: https://github.com/bcgsc/straglr

#### STR detection from short reads

ExpansionHunter: https://github.com/Illumina/ExpansionHunter

GangSTR: https://github.com/gymreklab/GangSTR

lobSTR: https://github.com/mgymrek/lobstr-code

HipSTR: https://github.com/tfwillems/HipSTR

## DNA modification detection

remora: https://github.com/nanoporetech/remora

DeepMod: https://github.com/WGLab/DeepMod

DeepMod2: https://github.com/WGLab/DeepMod2

DeepSignal: https://github.com/bioinfomaticsCSU/deepsignal

DeepSignal2: https://github.com/PengNi/deepsignal2

mCaller: https://github.com/al-mcintyre/mCaller

## Genome assembly and polishing

### Genome assembly

canu: https://github.com/marbl/canu

hifiasm: https://github.com/chhylp123/hifiasm

Flye: https://github.com/fenderglass/Flye

NextDenovo: https://github.com/Nextomics/NextDenovo

shasta: https://github.com/paoloshasta/shasta

wtdbg2: https://github.com/ruanjue/wtdbg2

NECAT: https://github.com/xiaochuanle/NECAT

### Genome polishing (error correction)

racon: https://github.com/isovic/racon

nanopolish: https://github.com/jts/nanopolish

medaka: https://github.com/nanoporetech/medaka

